Katana/Interceptor/ObserverInterceptor.swift

```diff
@@ -183,7 +195,7 @@ private struct ObserverLogic {
     with name: NSNotification.Name,
     _ typesToDispatch: [NotificationObserverDispatchable.Type]) {
     
-    NotificationCenter.default.addObserver(
+    self.notificationCenter.addObserver(
```

Comment:
Shouldn't we find a way to remove the observers when the Store is deallocated? I've seen that it exists this `NotificationCenter.default.removeObserver(_ observer: Any, name: Notification.Name, object: Any?)`. This requires keeping track of the observers we register and their notification.

It can also mean that the `ObserverInterceptor` should be a `class` and not a `struct` so that we can deregister them in the `deinit` without exposing any other weird stuff. 
Do you see any shortcomings? I think that we never pass around the `ObserverInterceptor` and for each Store there is only one `ObserverInterceptor` so it should be safe to change it to a class...